### ISSUE TITLE
Wrong line width when setting pen style in MAP_GRID_PEN

### ISSUE THREAD

**Issue by ZMAlt** [GenericMappingTools/gmt#8647]
The original post is [here](https://forum.generic-mapping-tools.org/t/pen-style-not-work-for-map-grid-pen/5656).

```
gmt basemap -Rd -JQ5i -Bafg --MAP_GRID_PEN=4p,black,dotted  -png test1
gmt basemap -Rd -JQ5i -Bafg --MAP_GRID_PEN=4p,black         -png test2
```
The first line gives:
![test1](https://github.com/user-attachments/assets/2248287b-b6cc-44d4-9e29-aae60478e20e)

The second line gives:
![test2](https://github.com/user-attachments/assets/72dd8b89-0301-4dda-b7eb-7b781e564a06)

**System information**

* Operating system: macos
* GMT version (`gmt --version`): gmt 6.5

**Comment by joa-quim** [comment]
This is a modern mode issue. In classic we get
```
psbasemap -Rd -JQ5i -Bafg --MAP_GRID_PEN=4p,black,dotted -P > lixo.ps
```
![lixo](https://github.com/user-attachments/assets/dce9b265-811c-47bd-8c06-1b88a62c7c3e)

**Comment by Esteban82** [comment]
I think that "it is not a bug it is a feature". See [comment on the forum ](https://forum.generic-mapping-tools.org/t/pen-style-not-work-for-map-grid-pen/5656/4?u=esteban82).

**Comment by joa-quim** [comment]
No, it is a bug nevertheless. And one very boring to catch. Tried a bit but gave up.

**Comment by joa-quim** [comment]
@ZMAlt  PR #9043 may have solved this (it did for the current example) but maybe there is more. It would be good if could test it a bit more

**Comment by ZMAlt** [comment]
It also works in other examples. Thank you.
